system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -43,8 +42,53 @@ ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSet
     PeriodicMetricSink(name, "elastic", pSettingsTree)
 {
     ignoreZeroMetrics = pSettingsTree->getPropBool("@ignoreZeroMetrics", true);
-    pSettingsTree->getProp("@elasticHost", elasticHost);
-    pSettingsTree->getProp("@indexName", indexName);
+
+    StringBuffer hostName;
+    StringBuffer hostProtocol;
+    StringBuffer hostPort;
+
+    Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
+
+    pHostConfigTree->getProp("@name", hostName);
+
+    if (!pHostConfigTree->getProp("@protocol", hostProtocol))
+    {
+        hostProtocol.append("https");
+    }
+
+    if (!pHostConfigTree->getProp("@port", hostPort))
+    {
+        hostPort.append("9200");
+    }
+
+    if (!hostName.isEmpty() && !hostPort.isEmpty() && !hostProtocol.isEmpty())
+    {
+        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+    }
+    else
+    {
+        WARNLOG("ElasticMetricSink: Host configuration is invalid");
```

Comment:
this commit doesn't show me how the sink will behave if we fall into this case. I see there's a "configurationValid" variable, but what happens when it's set to false? Does the sink simply refuse to report metrics?

Reply:
Understood. I think the question is, do we want to fail the load process and force admins to fix the misconfiguration immediately, or allow the components to load and report the error in logs. There's arguments for both, but if your sink cannot operate w/ the given misconfiguration we might consider error'ing out at load time.

Reply:
I am of the belief that we do not fail the load process. We log the appropriate warning and just don't collect metrics.